pkg/operator/encryptionprovider/encryptionprovider.go

```diff
@@ -0,0 +1,93 @@
+package encryptionprovider
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/operator/encryption/controllers"
+	"github.com/openshift/library-go/pkg/operator/encryption/encryptionconfig"
+	"github.com/openshift/library-go/pkg/operator/events"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+type encryptionProvider struct {
+	oauthAPIServerTargetNamespace   string
+	oauthEncryptionCfgAnnotationKey string
+
+	currentEncryptedGRs                 []schema.GroupResource
+	defaultEncryptedGRs                 []schema.GroupResource
+	encryptedGRsManagedByExternalServer sets.String
+
+	secretLister corev1listers.SecretNamespaceLister
+
+	eventRecorder events.Recorder
+}
+
+var _ controllers.Provider = &encryptionProvider{}
+
+func NewEncryptionProvider(
+	oauthAPIServerTargetNamespace string,
+	oauthEncryptionCfgAnnotationKey string,
+	defaultEncryptedGRs []schema.GroupResource,
+	encryptedGRsManagedByExternalServer sets.String,
+	kubeInformersForNamespaces operatorv1helpers.KubeInformersForNamespaces,
+	eventRecorder events.Recorder) *encryptionProvider {
+	return &encryptionProvider{
+		oauthAPIServerTargetNamespace:       oauthAPIServerTargetNamespace,
+		oauthEncryptionCfgAnnotationKey:     oauthEncryptionCfgAnnotationKey,
+		defaultEncryptedGRs:                 defaultEncryptedGRs,
+		encryptedGRsManagedByExternalServer: encryptedGRsManagedByExternalServer,
+		secretLister:                        kubeInformersForNamespaces.InformersFor(operatorclient.GlobalMachineSpecifiedConfigNamespace).Core().V1().Secrets().Lister().Secrets(operatorclient.GlobalMachineSpecifiedConfigNamespace),
+		eventRecorder:                       eventRecorder,
+	}
+}
+
+// EncryptedGRs returns resources that need to be encrypted
+func (p *encryptionProvider) EncryptedGRs() []schema.GroupResource {
+	oauthAPIServerEncryptionCfg, err := p.secretLister.Get(fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, p.oauthAPIServerTargetNamespace))
+	if err != nil {
+		return p.defaultEncryptedGRs
```

Comment:
I don't like this. If it is a NotFound, I agree with this. Can we be sure there are no other errors? Just returning the default list feels unsafe.

Reply:
Also note that before encryption is enabled, there is no secret. We would return an incomplete list in 4.5 here.

Reply:
oh wait, the default list is all? This is a bad name.

Reply:
what do you want to return on errors?
when can lister return an error?

returning the previously observed state can cause flapping

Reply:
that's why I would prefer to set it once on startup - now the value can change during runtime and it shouldn't.

Reply:
> oh wait, the default list is all? This is a bad name.

oh man, It's the default list this operator starts with :)
would you like me to rename to `allEncryptedGRs` ?